crates/primitives/src/context.rs

```diff
@@ -20,6 +20,12 @@ impl From<[u8; 32]> for ContextId {
     }
 }
 
+impl From<String> for ContextId {
+    fn from(s: String) -> Self {
+        s.parse().expect("Invalid ContextId string")
+    }
+}
```

Comment:
panicking is discouraged, and calling `.parse()` at source is the better approach anyway

Reply:
will fix it, still fighting with the compile error above.